packages/app-project/stores/Recents.spec.js

```diff
@@ -0,0 +1,204 @@
+import { expect } from 'chai'
+import sinon from 'sinon'
+import { getSnapshot } from 'mobx-state-tree'
+import auth from 'panoptes-client/lib/auth'
+
+import initStore from './initStore'
+
+describe('Stores > Recents', function () {
+  let rootStore
+  const project = {
+    id: '2',
+    display_name: 'Hello',
+    slug: 'test/project'
+  }
+
+  before(function () {
+    const mockResponse = {
+      body: {
+        recents: [
+          {
+            locations: [
+              { 'image/jpeg': 'subject.jpeg' }
+            ],
+            links: {
+              subject: '345'
+            }
+          }
+        ]
+      }
+    }
+    rootStore = initStore(true, { project })
+    sinon.stub(rootStore.client.panoptes, 'get').callsFake(() => Promise.resolve(mockResponse))
+  })
+
+  after(function () {
+    rootStore.client.panoptes.get.restore()
+  })
+
+  it('should exist', function () {
+    expect(rootStore.recents).to.be.ok
+  })
+
+  describe('with a project and user', function () {
+    before(function () {
+      const user = {
+        id: '123',
+        login: 'test.user'
+      }
+      sinon.stub(rootStore.collections, 'fetchFavourites')
+      rootStore.user.set(user)
+    })
+
+    after(function () {
+      rootStore.client.panoptes.get.resetHistory()
+      rootStore.collections.fetchFavourites.restore()
+    })
+
+    it('should request recent subjects from Panoptes', function () {
+      const token = 'Bearer '
+      const endpoint = '/users/123/recents'
+      const query = {
+        project_id: '2',
+        sort: '-created_at'
+      }
+      expect(rootStore.client.panoptes.get).to.have.been.calledOnceWith(endpoint, query, token)
+    })
+
+    it('should store existing recents', function () {
+      expect(rootStore.recents.recents).to.have.lengthOf(1)
+      const recents = getSnapshot(rootStore.recents.recents)
```

Comment:
`rootStore.recents.recents` is a bit hard to initially parse. Maybe this should be `rootStore.recentsStore.recents` or `rootStore.recents.resources`?

Reply:
MST has a `toJSON()` method that's helpful for testing too... so `rootStore.recents.toJSON()` should work.

Reply:
I might change that to `recentsStore.recents`.

Is there any difference in MST between `getSnapshot(store)` and `store.toJSON()`?

Reply:
Nope, they're the same (except you don't have to import `getSnapshot`) - https://github.com/mobxjs/mobx-state-tree#tojson-for-debugging